elasticmagic/src/commonMain/kotlin/dev/evo/elasticmagic/query/HasParent.kt

```diff
@@ -0,0 +1,32 @@
+package dev.evo.elasticmagic.query
+
+import dev.evo.elasticmagic.Params
+import dev.evo.elasticmagic.compile.BaseSearchQueryCompiler
+import dev.evo.elasticmagic.serde.Serializer
+
+data class HasParent(
+    val query: QueryExpression,
+    val parentType: String,
+    val score: Boolean = false,
```

Comment:
I prefer that optional parameters should be `nullable` to not duplicate Elasticsearch logic with default values.